### ISSUE TITLE
Allow setting consistency mode in Consul Namer and Dtab Store

### ISSUE THREAD

**Issue by Ashald** [BuoyantIO/linkerd#665]
As per https://www.consul.io/docs/agent/http.html, there are 3 consistency modes in Consul:
1) default
2) consistent
3) stale

It looks like that using default consistency mode with high rate of HTTP requests can lead to extra leader re-elections (https://groups.google.com/forum/#!msg/consul-tool/NrRRxem0Q_g/H9GFHRH3GQAJ)

The proposal is to add an extra option to allow tuning consistency mode for namer and dtab store.

I see 2 possible options:
1)
```
consistency: default/consistent/stale
```
2)
```
allowStale: true/false
```

**Comment by JustinVenus** [comment]
@Ashald if I get to vote I'd prefer option `1` and nice find on the leader election behavior was beginning to think I was seeing a consul bug.

**Comment by Ashald** [comment]
@JustinVenus yeah, that's exactly how it's being implemented - please check out #666

**Comment by olix0r** [comment]
Fixed by #666
